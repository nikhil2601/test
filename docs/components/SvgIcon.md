[pep-comp](/) > [docs](/docs/README.md) > [components](/docs/components/README.md) > SvgIcon

--------------------------------------------------------------------------------

# SvgIcon

**src**: [SvgIcon.js](/src/lib/SvgIcon/SvgIcon.js)

The `SvgIcon` component takes in an SVG `path` element and converts it to a React element that displays the path.

## Example

The following example is taken directly from the `[TrashIcon](/src/lib/SvgIcon/types/TrashIcon.js)` component.

```jsx
import React from 'react';
import PropTypes from 'prop-types';

import SvgIcon from '../SvgIcon';

const TrashIcon = ({ ...rest }) => (
    <SvgIcon {...rest}>
        <path d="M19,4H15.5L14.5,3H9.5L8.5,4H5V6H19M6,19A2,2 0 0,0 8,21H16A2,2 0 0,0 18,19V7H6V19Z" />
    </SvgIcon>
);

TrashIcon.propTypes = {
    viewBox: PropTypes.string
};

TrashIcon.defaultProps = {
    viewBox: '0 0 24 24'
};

export default TrashIcon;
```

## Props

```javascript
/**
 * Children passed into the SvgIcon.
 */
children: PropTypes.node,
/**
 * Set the color of the SvgIcon.
 */
color   : PropTypes.oneOfType([
    PropTypes.oneOf(['dark', 'error', 'inherit', 'light', 'primary', 'success', 'warning']),
    PropTypes.string
]),
/**
 * Provides a human-readable title for the element that contains it.
 * https://www.w3.org/TR/SVG-access/#Equivalent
 */
titleAccess: PropTypes.string,
/**
 * The viewBox attribute defines the position and dimension,
 * in user space, of an SVG viewport.
 * The value of the viewBox attribute is a list of four numbers:
 * min-x, min-y, width and height, separated by whitespace and/or a comma,
 * which specify a rectangle in user space which is mapped to the bounds
 * of the viewport established for the associated element.
 * https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/viewBox
 */
viewBox    : PropTypes.string
```

## Default Props

```javascript
children   : null,
color      : 'dark',
titleAccess: null,
viewBox    : '0 0 24 24'
```
