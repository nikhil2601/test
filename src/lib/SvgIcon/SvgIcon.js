import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { themeGet } from 'utils/theme';
import { renderPaletteColor } from 'utils/css';
import { renderStyle } from 'utils/styled';
import { responsiveProptypes } from 'utils/proptypes';

const SvgIcon = styled.svg.attrs(props => ({
    'aria-hidden': props.titleAccess ? 'false' : 'true',
    focusable: 'false',
}))`
    /**
     * Add all of the remaining styles from theme
     */
    ${themeGet('SvgIcon.styles')};
    /**
     * Add in dynamic styles
     */
    ${({ color, currentColor, height, margin, opacity, padding, theme, width, extraCss }) => css`
        display: inline-block;
        fill: currentColor;
        flex-shrink: 0;
        transition: all 1s ease-in-out;
        user-select: none;
        ${renderStyle(
            'color',
            color || 'inherit',
            theme,
            false,
            renderPaletteColor('color', 'color')
        )};
        ${renderStyle('height', height, theme)};
        ${renderStyle('margin', margin, theme)};
        ${renderStyle('opacity', opacity, theme)};
        ${renderStyle('padding', padding, theme)};
        ${renderStyle('width', width, theme)};
    `};

    ${({ extraCss }) => {
        return {
            ...extraCss,
        };
    }}
`;

SvgIcon.propTypes = {
    /**
     * Children passed into the SvgIcon.
     */
    children: PropTypes.node,
    /**
     * Set the color of the SvgIcon.
     */
    color: PropTypes.string,
    /**
     * Defines the 'height' style property.
     */
    height: responsiveProptypes(PropTypes.string),
    /**
     * Defines the 'margin' style property.
     */
    margin: responsiveProptypes(PropTypes.string),
    /**
     * Defines the 'opacity' style property.
     */
    opacity: responsiveProptypes(PropTypes.number),
    /**
     * Defines the 'padding' style property.
     */
    padding: responsiveProptypes(PropTypes.string),
    /**
     * Provides a human-readable title for the element that contains it.
     * https://www.w3.org/TR/SVG-access/#Equivalent
     */
    titleAccess: PropTypes.string,
    /**
     * The viewBox attribute defines the position and dimension,
     * in user space, of an SVG viewport.
     * The value of the viewBox attribute is a list of four numbers:
     * min-x, min-y, width and height, separated by whitespace and/or a comma,
     * which specify a rectangle in user space which is mapped to the bounds
     * of the viewport established for the associated element.
     * https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/viewBox
     */
    viewBox: PropTypes.string,
    /**
     * Defines the 'width' style property.
     */
    width: responsiveProptypes(PropTypes.string),
};

SvgIcon.defaultProps = {
    children: null,
    color: 'inherit',
    height: '1rem',
    margin: null,
    opacity: null,
    padding: null,
    titleAccess: null,
    viewBox: '0 0 24 24',
    width: '1rem',
};

export default SvgIcon;
