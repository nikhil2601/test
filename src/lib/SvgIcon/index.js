import SvgIcon from './SvgIcon';

// Import in the rest of the icons
import AlertIcon from './types/AlertIcon';
import ArrowIcon from './types/ArrowIcon';
import CheckboxEmptyIcon from './types/CheckboxEmptyIcon';
import CheckboxIcon from './types/CheckboxIcon';
import CheckboxIndeterminateIcon from './types/CheckboxIndeterminateIcon';
import ChevronIcon from './types/ChevronIcon';
import ClearIcon from './types/ClearIcon';
import EmailIcon from './types/EmailIcon';
import EyeIcon from './types/EyeIcon';
import GearIcon from './types/GearIcon';
import GridIcon from './types/GridIcon';
import KeyIcon from './types/KeyIcon';
import MenuIcon from './types/MenuIcon';
import PhoneIcon from './types/PhoneIcon';
import SearchIcon from './types/SearchIcon';
import TrashIcon from './types/TrashIcon';

/**
 * All of the SVG Icon types map with an easier to read key names.
 *
 * @type {Object}
 */
const types = {
    alert: AlertIcon,
    arrow: ArrowIcon,
    checkbox: CheckboxIcon,
    checkboxEmpty: CheckboxEmptyIcon,
    checkboxIndeterminate: CheckboxIndeterminateIcon,
    chevron: ChevronIcon,
    clear: ClearIcon,
    email: EmailIcon,
    eye: EyeIcon,
    gear: GearIcon,
    grid: GridIcon,
    key: KeyIcon,
    menu: MenuIcon,
    phone: PhoneIcon,
    search: SearchIcon,
    trash: TrashIcon,
};

/**
 * Add the types to the SVG Icon default export.
 *
 * @type {Object}
 */
SvgIcon.types = types;

// Default export the SvgIcon
export default SvgIcon;

export {
    AlertIcon,
    ArrowIcon,
    CheckboxEmptyIcon,
    CheckboxIcon,
    CheckboxIndeterminateIcon,
    ChevronIcon,
    ClearIcon,
    EmailIcon,
    EyeIcon,
    GearIcon,
    KeyIcon,
    MenuIcon,
    PhoneIcon,
    SearchIcon,
    GridIcon,
    TrashIcon,
};
