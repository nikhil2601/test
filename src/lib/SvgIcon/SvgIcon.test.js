import React from 'react';
import { shallow } from 'enzyme';

import SvgIcon from './index';

describe('<SvgIcon />', () => {
    const wrapper = shallow(<SvgIcon />);

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
