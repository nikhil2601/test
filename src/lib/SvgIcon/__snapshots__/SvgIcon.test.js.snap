// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgIcon /> should render correctly 1`] = `
<StyledComponent
  color="inherit"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        [Function],
      ],
      "componentStyle": ComponentStyle {
        "componentId": "SvgIcon-clbnd4-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          ";",
          [Function],
          ";",
          [Function],
        ],
      },
      "defaultProps": Object {
        "children": null,
        "color": "inherit",
        "height": "1rem",
        "margin": null,
        "opacity": null,
        "padding": null,
        "titleAccess": null,
        "viewBox": "0 0 24 24",
        "width": "1rem",
      },
      "displayName": "SvgIcon",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "children": [Function],
        "color": [Function],
        "height": [Function],
        "margin": [Function],
        "opacity": [Function],
        "padding": [Function],
        "titleAccess": [Function],
        "viewBox": [Function],
        "width": [Function],
      },
      "render": [Function],
      "styledComponentId": "SvgIcon-clbnd4-0",
      "target": "svg",
      "toString": [Function],
      "types": Object {
        "alert": [Function],
        "arrow": [Function],
        "checkbox": [Function],
        "checkboxEmpty": [Function],
        "checkboxIndeterminate": [Function],
        "chevron": [Function],
        "clear": [Function],
        "email": [Function],
        "eye": [Function],
        "gear": [Function],
        "grid": [Function],
        "key": [Function],
        "menu": [Function],
        "phone": [Function],
        "search": [Function],
        "trash": [Function],
      },
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  height="1rem"
  margin={null}
  opacity={null}
  padding={null}
  titleAccess={null}
  viewBox="0 0 24 24"
  width="1rem"
/>
`;
